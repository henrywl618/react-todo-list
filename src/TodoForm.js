import React, {useState} from "react";

const TodoForm = ({submitAction, removeTodo,title,edit=false,id})=>{

    let [todoInput,setTodoInput] = useState("");

    const handleChange = (e)=>{
        setTodoInput(e.target.value);
    };

    const handleSubmit = (e)=>{
        e.preventDefault();
        edit ? submitAction(id,todoInput) : submitAction(todoInput);
        setTodoInput("");
    };

    return (
        <form>
            <h3>{title}</h3>
            <label htmlFor="todo">Enter a reminder: </label>
            <input type="text"name="todo" id="todo" onChange={handleChange} value={todoInput} role="input"/>
            <button onClick={handleSubmit}>Submit</button>
        </form>
    );
};

export default TodoForm;