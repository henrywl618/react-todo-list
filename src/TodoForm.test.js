import React from "react";
import {render} from "@testing-library/react";
import TodoForm from "./TodoForm";

it("renders without crashing", ()=>{
    render(<TodoForm />);
});

it("matches the snapshot", ()=>{
    const {asFragment} = render(<TodoForm />);
    expect(asFragment()).toMatchSnapshot();
});