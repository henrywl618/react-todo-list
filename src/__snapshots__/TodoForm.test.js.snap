// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <form>
    <h3 />
    <label
      for="todo"
    >
      Enter a reminder: 
    </label>
    <input
      id="todo"
      name="todo"
      role="input"
      type="text"
      value=""
    />
    <button>
      Submit
    </button>
  </form>
</DocumentFragment>
`;
